app/docs/0.10.x/proxy.md

```diff
@@ -578,11 +578,30 @@ Kong will send the request over HTTP/1.1, and set the following headers:
 
 - `Host: <your_upstream_host>`, as previously described in this document.
 - `Connection: keep-alive`, to allow for reusing the upstream connections
-- `X-Real-IP: <$proxy_add_x_forwarded_for>`, where `$proxy_add_x_forwarded_for`
-  is the variable bearing the same name provided by
-  [ngx_http_proxy_module][ngx-http-proxy-module].
+- `X-Real-IP: <$remote_addr>`, where `$remote_addr` is the variable bearing
+  the same name provided by
+  [ngx_http_core_module][ngx-remote-addr-variable]. Please note that the
+  `$remote_addr` is likely overridden by
+  [ngx_http_realip_module][ngx-http-realip-module].
+- `X-Forwarded-For: <address>`, where `<address>` is the the content of
+  `$realip_remote_addr` provided by the
+  [ngx_http_realip_module][ngx-http-realip-module] appended to the request
+  header with the same name.
 - `X-Forwarded-Proto: <protocol>`, where `<protocol>` is the protocol used by
-  the client.
+  the client. In the case where `$realip_remote_addr` is one of the **trusted**
+  addresses, the request header with the same name gets forwarded if provided.
+  Otherwise, the value of `$scheme` variable provided by the
+  [ngx_http_core_module][ngx-scheme-variable] will be used.
+- `X-Forwarded-Host: <host>`, where `<host>` is the host name sent by
+  the client. In the case where `$realip_remote_addr` is one of the **trusted**
+  addresses, the request header with the same name gets forwarded if provided.
+  Otherwise, the value of `$host` variable provided by the
+  [ngx_http_core_module][ngx-host-variable] will be used.
+- `X-Forwarded-Port: <port>`, where `<port>` is the port of the server which
+  accepted a request. In the case where `$realip_remote_addr` is one of the
+  **trusted** addresses, the request header with the same name gets forwarded
+  if provided. Otherwise, the value of `$server_port` variable provided by the
```

Comment:
Other mentions of Nginx module are not referred to with a `the` pronoun, which I think makes more sense. Mind removing the "the" here and on lines 598, 593, and 587 as well?